ores/api.py

```diff
@@ -81,22 +87,31 @@ def _score(self, context, models, rev_ids):
         logging.debug("Starting up thread pool with {0} workers"
                       .format(self.workers))
         with ThreadPoolExecutor(max_workers=self.workers) as executor:
-            futures = []  # A list of future results
+            future_rev_ids = []  # A list of future results and the revids
 
             # This loop loads all rev_id_batch's into the executor for
             # processing
             for rev_id_batch in chunked(rev_ids, self.batch_size):
                 rev_id_batch = list(rev_id_batch)
                 logging.debug("Starting batch of {0} revids"
                               .format(len(rev_id_batch)))
-                futures.append(executor.submit(self._score_request,
-                                               context, rev_id_batch,
-                                               models))
+                future_rev_ids.append((
+                    executor.submit(self._score_request, context, rev_id_batch,
+                                    models),
+                    rev_id_batch))
 
             # This loop blocks on reading the futures as soon as they are ready
-            for future in futures:
-                for score in future.result():
-                    yield score
+            for (future, rev_id_batch) in future_rev_ids:
+                try:
+                    for score in future.result():
+                        yield score
+                except RuntimeError as e:
+                    logger.warning(
+                        "An ORES scoring job failed with the following error:")
+                    logger.warning(traceback.format_exc())
+                    for rev_id in rev_id_batch:
+                        yield {m: {"error": e.args[0]}
```

Comment:
Responsibility more plausibly lies with `_score_request`, below.

Reply:
Good point.  {{done}}

Reply:
Just making sure I have this right—if we get a MediaWiki timeout error, it'll already be wrapped by lower-level functions and we won't encounter here, right?  It would be unfortunate to mark all revisions as errored.

I'm okay with the error map being built here though, at least it's in the right class.

Reply:
Right.  This is a limitation of our batching strategy.  When we're batching 50 revisions together, any API failure will cause the scoring of all 50 revisions to fail.